rollup.base.config.js

```diff
@@ -24,26 +24,124 @@ function doCleanup() {
   })
 }
 
-const browserRollupConfigFactory = (banner, importCheckNames, targetType, theNameSpace, entryInputName, outputName, libVersion, isProduction, format = 'umd', postfix = '') => {
+const getNamespace = (prefix, namespaces, baseName, rootName) => {
+    var result = prefix + "var " + baseName + "=" + rootName;
+    if (namespaces.length > 0) {
+        for (let lp = 0; lp < namespaces.length; lp++) {
+            if (lp === 0) {
+                result += ", ";
+            } else {
+                result += ";\n" + prefix;
+            }
+            result += "nsKey=\"" + namespaces[lp] + "\", ";
+            result += baseName + "=" + baseName + "[nsKey]=(" + baseName + "[nsKey]||{})";
+        }
+    }
+
+    return result + ";\n";
+}
+
+const getCommonNamespace = (browserNs, gblNs) => {
+    var brTokens = browserNs.split(".");
+    var gblTokens = gblNs.split(".");
+    let common = [];
+    let idx = 0;
+    while (brTokens.length > idx && gblTokens.length > idx && brTokens[idx] === gblTokens[idx]) {
+        common.push(brTokens[idx]);
+        idx++;
+    }
+
+    return {
+        common: common,
+        browser: brTokens.slice(idx),
+        gbl: gblTokens.slice(idx)
+    };
+};
+
+const getIntro = (format, theNameSpace, moduleName, theVersion) => {
```

Comment:
Simplistically, this uses the rollup `intro` and `outro` config options to "take control" of the prefix and postfix (the code rollup wraps around the generated files) so that we can more efficiently create multiple namespaces (Microsoft.ApplicationInsights and Microsoft.ApplicationInsights__X__ (where X is the major version).

This also allow us to add additional debugging information to assist with supporting environment that "load" multiple major versions.